File: questão2.js
const readline = require('readline');

const rl = readline.criandoInterface({
    input: process.stdin,
    output: process.stdout
});

rl.pergunte('Digite o dia (1-31): ', (dia) => {
    rl.pergunte('Digite o mês (1-12): ', (mes) => {
        dia = parseInt(dia, 10);
        mes = parseInt(mes, 10);

        function EstacaoAno(dia, mes) {
            if ([1, 2].includes(mes)) {
                return 'VERAO';
            } else if (mes === 3) {
                return dia < 21 ? 'VERAO' : 'OUTONO';
            } else if ([4, 5].includes(mes)) {
                return 'OUTONO';
            } else if (mes === 6) {
                return dia < 21 ? 'OUTONO' : 'INVERNO';
            } else if ([7, 8].includes(mes)) {
                return 'INVERNO';
            } else if (mes === 9) {
                return dia < 21 ? 'INVERNO' : 'PRIMAVERA';
            } else if ([10, 11].includes(mes)) {
                return 'PRIMAVERA';
            } else if (mes === 12) {
                return dia < 21 ? 'PRIMAVERA' : 'VERAO';
            }
        }

        if (1 <= dia && dia <= 31 && 1 <= mes && mes <= 12) {
            let estacao = EstacaoAno(dia, mes);
            console.log(`A estação para ${dia}/${mes} é ${estacao}.`);
        } else {
            console.log("Por favor, digite valores válidos para dia (1-31) e mês (1-12).");
        }
        
        rl.close();
    });
});

File: questão2.py
from datetime import datetime

estacoes = {
    'Verão':  {'inicio': (21, 12), 'fim': (20, 3)},
    'Outono': {'inicio': (21, 3), 'fim': (20, 6)},
    'Inverno':{'inicio': (21, 6), 'fim': (22, 9)},
    'Primavera':{'inicio': (23, 9), 'fim': (20, 12)}
}

def get_estacao(data_nascimento):
    dia, mes = data_nascimento.day, data_nascimento.month
    for estacao, periodo in estacoes.items():
        inicio = datetime(datetime.now().year, periodo['inicio'][1], periodo['inicio'][0])
        fim = datetime(datetime.now().year, periodo['fim'][1], periodo['fim'][0])

        if inicio <= data_nascimento <= fim:
            return estacao
        if periodo['inicio'][1] > periodo['fim'][1] or (periodo['inicio'][1] == periodo['fim'][1] and periodo['inicio'][0] > periodo['fim'][0]):
            if data_nascimento >= inicio or data_nascimento <= fim:
                return estacao
    return "Data inválida"

data_nascimento_str = input("Informe sua data de nascimento (DD/MM): ")
data_nascimento = datetime.strptime(data_nascimento_str, "%d/%m")

estacao_aniversario = get_estacao(data_nascimento)
print(f"Você nasceu na estação: {estacao_aniversario}")
